### PR TITLE
Fix Build Issues + Add backup mod name

### PR DESCRIPTION
**Pull request by IntegerLimit**
A backup tooltip has been implemented, meaning that if it is unable to get the Mod Container, or the Mod Name, it will return the item's registry name's domain, capitalized. (Pretty much same as TOP's system)

Also, issues with building the mod have been fixed, and gradle updated to 4.1.0.

### CODE REVIEW

File: src/main/java/mezz/modnametooltip/TooltipEventHandler.java [hunk 1]
```diff
@@ -33,6 +38,14 @@ public void onToolTip(ItemTooltipEvent event) {
 	private static String getModName(ItemStack itemStack) {
 		if (!itemStack.isEmpty()) {
 			Item item = itemStack.getItem();
+
+			// Override for Groovyscript
+			if (ModNameTooltip.isGroovyLoaded()){
+				if (Objects.requireNonNull(item.getRegistryName()).getResourceDomain().equals(
+						GroovyScript.getRunConfig().getPackId()))
+					return GroovyScript.getRunConfig().getPackName();
+			}
```
Comment (mezz) [inline src/main/java/mezz/modnametooltip/TooltipEventHandler.java]: I'd like to hide away this logic in a file like `GroovyScriptHack.java` and then call it here like
```suggestion
			String groovyScriptName = GroovyScriptHack.getName(item);
			if (groovyScriptName != null) {
				return groovyScriptName;
			}

```

File: src/main/java/mezz/modnametooltip/TooltipEventHandler.java [hunk 2]
```diff
@@ -33,6 +38,14 @@ public void onToolTip(ItemTooltipEvent event) {
 	private static String getModName(ItemStack itemStack) {
 		if (!itemStack.isEmpty()) {
 			Item item = itemStack.getItem();
+
+			// Override for Groovyscript
+			if (ModNameTooltip.isGroovyLoaded()){
+				if (Objects.requireNonNull(item.getRegistryName()).getResourceDomain().equals(
+						GroovyScript.getRunConfig().getPackId()))
+					return GroovyScript.getRunConfig().getPackName();
```
Comment (mezz) [inline src/main/java/mezz/modnametooltip/TooltipEventHandler.java]: Can we hard-code these strings or ids and remove the dependency on GroovyScript?

**Review by mezz — CHANGES_REQUESTED** [review]

File: src/main/java/mezz/modnametooltip/TooltipEventHandler.java (hunk 2 above)
Comment (IntegerLimit) [reply src/main/java/mezz/modnametooltip/TooltipEventHandler.java]: Sadly, these can't be hardcoded, as groovy allows editing of the pack name and pack id. As an example, one modpack might have a pack id (the domain of the registry name) of `tj` and a pack name of `Technological Journey`, whilst another has `nomifactory` as the domain and `Nomifactory` as the pack name.

File: src/main/java/mezz/modnametooltip/TooltipEventHandler.java (hunk 1 above)
Comment (IntegerLimit) [reply src/main/java/mezz/modnametooltip/TooltipEventHandler.java]: Ok, I'll do this later today.

Comment (IntegerLimit) [reply src/main/java/mezz/modnametooltip/TooltipEventHandler.java]: Done. I've created a folder called compat, in case this happens with another mod, and further compat scripts can go there.
https://github.com/mezz/ModNameTooltip/pull/30/commits/19c35e9e72a8dff48841669ae186f8129a06395a

**Comment by IntegerLimit** [comment]
There might be another way, because JEI's tooltips work perfectly fine with GS's custom items & blocks. How does JEI get the mod name for their tooltips (In the main JEI viewing page)?

**Comment by IntegerLimit** [comment]
Note: Getting the mod container, with the domain of the resource location or the creator mod id, won't work, as GS doesn't have a mod container.

**Comment by mezz** [comment]
JEI's mod name tooltip code for 1.12 is in this file, starting here:
https://github.com/mezz/JustEnoughItems/blob/eac5cdab028a16395134eb1cab305480ff04ec8c/src/main/java/mezz/jei/startup/ForgeModIdHelper.java#L104-L117

**Comment by mezz** [comment]
The code is a bit hard to follow in JEI, so I will help here:

I get the display mod id from the ItemStack here:
https://github.com/mezz/JustEnoughItems/blob/eac5cdab028a16395134eb1cab305480ff04ec8c/src/main/java/mezz/jei/plugins/vanilla/ingredients/item/ItemStackHelper.java#L95-L106

Then the mod id is turned into a mod name here:
https://github.com/mezz/JustEnoughItems/blob/eac5cdab028a16395134eb1cab305480ff04ec8c/src/main/java/mezz/jei/startup/ForgeModIdHelper.java#L44-L51

**Comment by IntegerLimit** [comment]
Ok, I'll have a look at it and see if I can fix it, by following how JEI does it.

**Comment by IntegerLimit** [comment]
At a glance, the code looks... pretty much exactly the same. I will further see if there is any difference.

**Comment by IntegerLimit** [comment]
Somehow, the code looks pretty much the same for JEI, using the same methods, and the same way to retrieve the mod name. Perhaps it has to do with the IndexedModList being grabbed before the tooltip event? I have no idea, but I'll see if retrieving the IndexedModList in PreInit fixes the problem. If it does, I'll change the PR, but keep the backup.

**Comment by IntegerLimit** [comment]
There is absolutely no difference, I have zero idea why JEI's tooltips works but this mod's doesn't.

**Comment by IntegerLimit** [comment]
Found the problem. GS uses Mixin to edit JEI, so that it displays the correct mod name. Reverting the GS related stuff, as this will most likely be fixed GS-side.

**Comment by IntegerLimit** [comment]
Ok, now this pr simply fixes build problems and adds a backup mod name to use.

File: build.gradle [hunk 3]
```diff
@@ -1,21 +1,33 @@
...
 archivesBaseName = "modnametooltip_" + mcversion
 
+repositories {
+    maven {
+        url "https://cursemaven.com"
+    }
+}
+
+dependencies {
+    compile "curse.maven:groovyscript-687577:4457579"
```
Comment (mezz) [inline build.gradle:28]: this dependency is no longer needed

**Review by mezz — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 3 above)
Comment (IntegerLimit) [reply build.gradle:28]: Yep, forgot that, thanks, changing now.

**Comment by IntegerLimit** [comment]
Also, in case your wondering, the original issue has now been fixed GS side. I think the backup mod name is still worthwhile though, as this can happen with other mods.
